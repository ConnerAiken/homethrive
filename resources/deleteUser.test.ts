const deleteUser = require('./../resources/deleteUser');

describe('Delete user actions DELETE->/users/', () => {
    const baseRequest = {
        resource: '/users',
        httpMethod: 'DELETE',
    };

    test('We are able to delete a user using the DELETE endpoint', async () => {
        const requestPayload = {
            ...baseRequest,
            path: '/users',
            body: JSON.stringify({
                id: '1',
            }),
        };

        const responsePayload = await deleteUser(requestPayload, {
            delete: jest.fn().mockReturnValue({
                promise: () => {
                    return {
                        Attributes: {
                            id: '1',
                            firstName: 'Conner',
                            lastName: 'Aiken',
                            email: 'conner@fittedtech.com',
                        },
                    };
                },
            }),
        });

        expect(responsePayload.statusCode).toBe(200);
        expect(responsePayload.body).toContain('Successfully deleted');
        expect(JSON.parse(responsePayload.body)).toMatchSnapshot();
    });

    test('We send the proper server status code when there is an invalid body JSON', async () => {
        const requestPayload = {
            ...baseRequest,
            path: '/users',
            body: 'invalid json',
        };

        const responsePayload = await deleteUser(requestPayload, {
            get: jest.fn(),
        });

        expect(responsePayload.statusCode).toBe(400);
        expect(responsePayload.body).toContain('Invalid JSON body');
    });

    test('We send the proper server status code when we cannot find the record to delete', async () => {
        const requestPayload = {
            ...baseRequest,
            path: '/users',
            body: JSON.stringify({
                id: '123',
            }),
        };

        const responsePayload = await deleteUser(requestPayload, {
            delete: jest.fn().mockReturnValue({
                promise: () => {
                    return {};
                },
            }),
        });

        expect(responsePayload.statusCode).toBe(404);
        expect(responsePayload.body).toContain(
            'Could not find the relevant user'
        );
        expect(responsePayload.body).toMatchSnapshot();
    });

    test('We handle generic errors properly', async () => {
        const requestPayload = {
            ...baseRequest,
            path: '/users',
            body: JSON.stringify({
                id: '1',
            }),
        };

        const responsePayload = await deleteUser(requestPayload, {
            delete: jest.fn().mockReturnValue({
                promise: () => {
                    throw new Error('Simulated error');
                },
            }),
        });

        expect(responsePayload.statusCode).toBe(500);
        expect(responsePayload.body).toContain('An error occured');
        expect(responsePayload.body).toMatchSnapshot();
    });
});
