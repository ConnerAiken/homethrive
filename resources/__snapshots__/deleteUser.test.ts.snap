// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete user actions DELETE->/users/ We are able to delete a user using the DELETE endpoint 1`] = `
{
  "error": false,
  "message": "Successfully deleted",
  "payload": {
    "email": "conner@fittedtech.com",
    "firstName": "Conner",
    "id": "1",
    "lastName": "Aiken",
  },
}
`;

exports[`Delete user actions DELETE->/users/ We are able to get a user using the GET endpoint 1`] = `
{
  "error": false,
  "message": "Successfully deleted",
  "payload": {
    "email": "conner@fittedtech.com",
    "firstName": "Conner",
    "id": "1",
    "lastName": "Aiken",
  },
}
`;

exports[`Delete user actions DELETE->/users/ We handle generic errors properly 1`] = `"{"error":{},"message":"An error occured","payload":{}}"`;

exports[`Delete user actions DELETE->/users/ We send the proper server status code when we cannot find the record to delete 1`] = `"{"message":"Could not find the relevant user to delete","error":true,"payload":{}}"`;
